### PR TITLE
N+1 problems

### PR DESCRIPTION
**Pull request by kosher9**
Changes made in this pull request : 
- [x] Fixed the n+1 problem.
- [x] Added spec test for user#index and user#show
Screenshot_20220719_141016

- [x] Added spec test for post#index and post#show
Screenshot_20220719_141216

### CODE REVIEW

File: spec/features/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        expect(page).to have_content('John')
+      end
+
+      scenario 'I can see the number of posts the user has written' do
+        expect(page).to have_content("Number of posts : #{@first_user.posts_counter}")
+      end
+
+      scenario 'I can see the user\'s bio' do
+        expect(page).to have_content('Developer.')
+      end
```
Comment (TedLivist) [inline spec/features/user_spec.rb]: - [ ] For the user show page you are required to add tests for the scenarios listed below. Kindly add them.

  - Test that redirects to the post's show page when you click on a user's post.
  - Test that redirects to the user's post's index page when you click on a see all posts.

File: spec/features/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+RSpec.describe Post, type: :model do
```
Comment (TedLivist) [inline spec/features/posts_spec.rb]: - [x] I believe we are testing the `feature` type here, not `model`. Kindly fix that.

File: spec/features/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    it 'if comments counter greater than or equal to zero.' do
+      @post.comments_counter = -5
+      expect(@post).to_not be_valid
+    end
+    it 'if comments counter is integer' do
+      @post.comments_counter = 8
+      expect(@post).to be_valid
+    end
+  end
+end
```
Comment (TedLivist) [inline spec/features/posts_spec.rb]: - [ ] These tests seem to be merely testing the model modifications, whereas you are required to write featured tests that visit the index page and check for different scenarios. Kindly refactor your tests to do that.

- [ ] Remember to include ALL scenarios for the Post #index page listed in the guidelines.

File: spec/features/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    it 'Can see the post body' do
+      expect(page).to have_content 'This is my first post'
+    end
+    it 'Can see the username of each commentor' do
+      expect(page).to have_content 'Nemwel'
+    end
+    it 'Can see the comment each commentor left' do
+      expect(page).to have_content 'My first comment'
+      expect(page).to have_content 'My second comment'
+    end
```
Comment (TedLivist) [inline spec/features/posts_spec.rb]: - [ ] You are missing the test for the post title. Kindly include that as well

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @kosher9 and @Nemwel-Boniface,

Good job so far!
✔️ Good documentation
✔️ Inclusion of `capybara` for tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] I was unable to find the **`featured`** tests for the login page in the `spec` directory. Kindly include them.

  login-spec

- [ ] **_Check the comments under the review._**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kosher9** [comment]
Hi @TedLivist, there is no way of testing the login page without having the login page. The materials have changed and Microverse has to fix that. Please, try to find out about it.

**Comment by TedLivist** [comment]
@kosher9 you are right. Login test is not part of this project. Kindly ignore that request.

**Review by Lordkaito — APPROVED** [review]
Hi @kosher9, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**